h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -2757,6 +2757,10 @@ private boolean updateEarlyStop() {
   
   private Solver defaultSolver() {
     Solver s = Solver.IRLSM;
+    if (_parms._remove_collinear_columns) { // choose IRLSM if remove_collinear_columns is true
+      _parms._solver = s;
+      return s;
```

Comment:
@wendycwong this early return is causing that the picked solver is not being logged, see: https://github.com/h2oai/h2o-3/pull/5419/files#diff-bfdd0a804deb47b13d4ca209feaa9c26d2845ef8192f44c23bc4d5ab6caac585R2779

Reply:
I added Log.info to log the new solver.  Is this what you are looking for?

Reply:
Good point.